diff --git a/guest-account.sh b/guest-account.sh
index 0321723..4f935c6 100644
--- a/guest-account.sh
+++ b/guest-account.sh
@@ -83,36 +83,23 @@ add_account ()
     }
   fi
 
-  site_gs=/etc/guest-session
-
   # create temporary home directory
   mount -t tmpfs -o mode=700,uid=${GUEST_USER} none ${GUEST_HOME} || {
     rm -rf ${GUEST_HOME}
     exit 1
   }
 
-  if [ -d ${site_gs}/skel ] && [ "$(ls -A ${site_gs}/skel)" ]; then
-    cp -rT ${site_gs}/skel/ ${GUEST_HOME}
-    chown -R ${GUEST_USER}:${GUEST_USER} ${GUEST_HOME}
+  if [ -d /etc/guest-session/skel ] && [ "$(ls -A /etc/guest-session/skel)" ]; then
+    cp -rT /etc/guest-session/skel/ ${GUEST_HOME}
+    chown -R ${GUEST_USER}:users ${GUEST_HOME}
   else
     cp -rT /etc/skel/ ${GUEST_HOME}
-    chown -R ${GUEST_USER}:${GUEST_USER} ${GUEST_HOME}
+    chown -R ${GUEST_USER}:users ${GUEST_HOME}
   fi
 
   # setup session
-  if [ -f ${site_gs}/setup.sh ]; then
-    su ${GUEST_USER} -c "env HOME=${GUEST_HOME} site_gs=${site_gs} ${site_gs}/setup.sh"
-  fi
-
-  # set possible local guest session preferences
-  source_local_prefs() {
-    local USER=${GUEST_USER}
-    local HOME=${GUEST_HOME}
-    . ${site_gs}/prefs.sh
-    chown -R ${USER}:${USER} ${HOME}
-  }
-  if [ -f ${site_gs}/prefs.sh ]; then
-    source_local_prefs
+  if [ -f /etc/guest-session/setup.sh ]; then
+    su ${GUEST_USER} -c "env HOME=${GUEST_HOME} site_gs=/etc/guest-session /etc/guest-session/setup.sh"
   fi
 
   echo ${GUEST_USER}
